### PR TITLE
add Transmittable persistance operations

### PR DESCRIPTION
**Pull request by jacobkagon**
https://www.pivotaltracker.com/n/projects/2640060/stories/187939070

### CODE REVIEW

File: app/domain/operations/transmittable/generate_request_objects.rb [hunk 1]
```diff
@@ -0,0 +1,138 @@
+# frozen_string_literal: true
+
+module Operations
```
Comment (saikumar9) [inline app/domain/operations/transmittable/generate_request_objects.rb]: @jacobkagon I recommend you to nest this under the SugarCRM namespace as this is not a generic operation.

File: app/domain/operations/transmittable/generate_request_objects.rb [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+      def call(params)
+        values = yield validate(params)
+        job_params = yield construct_job_params
+        @job = yield create_job(job_params)
+        transmission_params = yield construct_transmission_params(values)
+        @transmission = yield create_request_transmission(transmission_params, @job)
+        @family = yield create_family(values)
+        transaction_params = yield construct_transaction_params
+        @transaction = yield create_request_transaction(transaction_params, @job)
```
Comment (saikumar9) [inline app/domain/operations/transmittable/generate_request_objects.rb]: @jacobkagon We discussed having multiple transactions, one for the Account and one per contact. I do not see that here.

File: app/domain/operations/transmittable/generate_request_objects.rb [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+            subject: @family,
+            key: :family_created_or_updated,
+            title: "Family created or updated transaction for family: #{@family.family_hbx_id}.",
+            description: "Family created or updated transaction for family: #{@family.family_hbx_id}.",
+            publish_on: Time.zone.now,
+            started_at: DateTime.now,
+            event: 'initial',
+            state_key: :initial,
+            correlation_id: @family.family_hbx_id
+          }
```
Comment (saikumar9) [inline app/domain/operations/transmittable/generate_request_objects.rb]: I do not see the payload being persisted on the transaction here. We have to either save the account or contact related payload accordingly.

File: app/domain/operations/transmittable/generate_request_objects.rb [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        @transaction.xml_payload = payload if payload.instance_of?(String)
+        if @transaction.save
+          Success(@transaction)
+        else
+          add_errors(:save_payload, "Failed to save payload on response transaction", { job: @job, transmission: @transmission, transaction: @transaction })
+          status_result = update_status(result.failure, :failed, { job: @job, transmission: @transmission, transaction: @transaction })
+          return status_result if status_result.failure?
+          Failure("Failed to save payload on response transaction")
+        end
+      end
```
Comment (saikumar9) [inline app/domain/operations/transmittable/generate_request_objects.rb]: This method is not being used anywhere.

File: app/domain/operations/families/create.rb [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        else
+          Rails.logger.error { "Family creation failed: #{family.errors.full_messages}" }
+          Failure("Family creation failed")
+        end
+      end
+
+      def create_family_params(values)
+        Success(
+          {
+            correlation_id: values[:inbound_family_cv][:hbx_id],
```
Comment (saikumar9) [inline app/domain/operations/families/create.rb:40]: The correlation_id should be a unique secure random string generated and set from EA to Crm per my understanding. Please review and make changes if required.

File: spec/domain/operations/families/create_spec.rb [hunk 6]
```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Operations::Families::Create, dbclean: :after_each do
```
Comment (saikumar9) [inline spec/domain/operations/families/create_spec.rb:6]: Good test coverage.

**Review by saikumar9 — CHANGES_REQUESTED** [review]

File: spec/domain/operations/families/updated_or_created_processor_spec.rb [hunk 7]
```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Operations::Families::UpdatedOrCreatedProcessor, dbclean: :after_each do
+  let(:file_data) { File.read("spec/test_data/cv3_payload.json") }
+  let(:cv3_family_payload) { JSON.parse(JSON.parse(file_data), symbolize_names: true) }
+  let(:after_save_updated_at) { DateTime.now.to_s }
+
+  describe "Success" do
```
Comment (saikumar9) [inline spec/domain/operations/families/updated_or_created_processor_spec.rb:12]: @jacobkagon For the success case we can verify the below:
* Check if the persistence objects are being created properly.
* Check if we can navigate between objects (transmittable and the subject) using the associations properly.

**Review by saikumar9 — APPROVED** [review]
LGTM!

File: Gemfile.lock [hunk 8]
```diff
@@ -506,7 +506,7 @@ DEPENDENCIES
   rails (~> 7.0.8.4)
   rbtree (~> 0.4.5)
   resource_registry!
-  rexml (~> 3.2.7)
+  rexml (~> 3.3.2)
```
Comment (jayreddy519) [inline Gemfile.lock:509]: changed rexml version.

File: spec/domain/operations/sugar_crm/transmittable/generate_request_objects_spec.rb [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+    end
+
+    context "with invalid params" do
+
+      it "returns a failure response" do
+        expect(@result.failure?).to be_truthy
+      end
+
+      it "returns a failure message" do
+        expect(@result.failure).to eql("Inbound family cv payload not present")
```
Comment (jayreddy519) [inline spec/domain/operations/sugar_crm/transmittable/generate_request_objects_spec.rb:59]: good spec coverage for positive & negative cases.

**Review by jayreddy519 — APPROVED** [review]
lgtm.

**Pull request merged**